src/styles/main.scss

```diff
@@ -1,3 +1,59 @@
+$days-of-week: 30;
+
 body {
   margin: 0;
 }
+
+.calendar {
+  position: fixed;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: 780px;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+
+  &__day {
+    background-color: #eee;
+    width: 100px;
+    height: 100px;
+    border: 1px solid black;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transition: cursor, background-color, transform, 0.5s;
+  }
+
+  @for $i from 1 through $days-of-week {
+    &__day:nth-child(#{$i})::before {
+      content: #{$i};
+    }
+  }
+
+  @if $days-of-week == 28 {
+    &__day:nth-child(n + 29) {
+      display: none;
+    }
+  }
+
+  @else if $days-of-week == 29 {
+    &__day:nth-child(n + 30) {
+      display: none;
+    }
+  }
+
+  @else if $days-of-week == 30 {
+    &__day:nth-child(n + 31) {
+      display: none;
+    }
+  }
```

Comment:
Try to use `for` cycle here too

Reply:
ok, thank's!

Reply:
But I don't understand why is it needed?